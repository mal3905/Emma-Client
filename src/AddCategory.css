/* button {
    margin-top: 20px;
    color: black;
    text-decoration: none;
    
}

label {
    font-size: 20px;
    margin-right: 10px;
}

h2 {
    position: relative;

} */

