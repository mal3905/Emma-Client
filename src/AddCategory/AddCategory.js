import React, { Component } from 'react'
import ItemForm from '../ItemForm/ItemForm'
import ApiContext from '../ApiContext'
import config from '../config'




export  default class AddCategory extends Component {
    static defaultProps = {
        history: {
          push: () => { }
        },
      }

      static contextType = ApiContext;
     
      handleSubmit = e => {
        e.preventDefault()
        const category = {
          name: e.target['cat-name'].value
        }

        fetch(`${config.API_ENDPOINT}/category`, {
            method: 'POST',
            headers: {
              'content-type': 'application/json'
            },
            body: JSON.stringify(category),
          })
          .then(res => {
            if (!res.ok)
              return res.json().then(e => Promise.reject(e))
            return res.json()
          })
          .then(category => {
            this.context.addCategory(category)
            this.props.history.push(`/category/${category.id}`)
          })
          .catch(error => {
            console.error({ error })
          })
      }

    render() {
        return (
            <section className='AddFolder'>
            <h2>Create category</h2>
            <ItemForm onSubmit={this.handleSubmit}>
              <div className='field'>
                <label htmlFor='cat-name-input'>
                  Name
                </label>
                <input type='text' id='cat-name-input' name='cat-name' />
              </div>
              <div className='buttons'>
                <button type='submit'>
                  Add Category
                </button>
              </div>
            </ItemForm>
          </section>
        )
    }
}