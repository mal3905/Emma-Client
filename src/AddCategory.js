import React, { Component } from 'react'
import EmmaForm from './EmmaForm'
import MyContext from './MyContext'
import config from './config'
import './AddCategory.css'


export default class AddCategory extends Component {
  static defaultProps = {
    history: {
      push: () => { }
    },
  }
  static contextType = MyContext;

  handleSubmit = e => {
    e.preventDefault()
    const categories = {
      name: e.target['category-name-input'].value

    }
    // console.log(`for add ${categories}`)
    // console.log(this.name.value)
    // console.log()
    
    fetch(`${config.API_ENDPOINT}/category`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(categories),
    })
  
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      }) 
      .then(category => {
        this.context.addCategories({...categories, id: category.id})
        this.props.history.push(`/category/${category.id}`)
        // console.log(this.state)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    return (
      <section className='AddCategory'>
        <h2>Create Category</h2>
        <EmmaForm onSubmit={this.handleSubmit}>
          <div className='field'>
            <label htmlFor='category-name-input'>
             
            </label>
            <input type='text' id='category-name-input' name='category-name' placeholder='name'/>
          </div>
          <div className='buttons'>
            <button type='submit'>
              Add category
            </button>
          </div>
        </EmmaForm>
      </section>
    )
  }
}