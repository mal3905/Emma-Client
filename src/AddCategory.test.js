import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddCategory from './AddCategory'

describe(`AddItemForm component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<AddCategory />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
