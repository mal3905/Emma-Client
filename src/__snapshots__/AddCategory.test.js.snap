// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm component renders the complete form 1`] = `
<section
  className="AddCategory"
>
  <h2>
    Create Category
  </h2>
  <EmmaForm
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="category-name-input"
      />
      <input
        id="category-name-input"
        name="category-name"
        placeholder="name"
        type="text"
      />
    </div>
    <div
      className="buttons"
    >
      <button
        type="submit"
      >
        Add category
      </button>
    </div>
  </EmmaForm>
</section>
`;
